sm/plugin_sm/src/plugin.rs

```diff
@@ -65,13 +86,13 @@ impl ExternalPluginCommand {
         &self,
         action: &str,
         maybe_module: Option<&SoftwareModule>,
-    ) -> Result<Command, SoftwareError> {
+    ) -> Result<LoggedCommand, SoftwareError> {
         let mut command = if let Some(sudo) = &self.sudo {
-            let mut command = Command::new(&sudo);
-            command.arg(&self.path);
+            let mut command = LoggedCommand::new(sudo.to_str().unwrap());
+            command.arg(&self.path.to_str().unwrap());
             command
         } else {
-            Command::new(&self.path)
+            LoggedCommand::new(&self.path.to_str().unwrap())
```

Comment:
unwrapping because the new does not return any error?.

Reply:
This `unwrap()` is definitely unsafe since the path is read from the filesystem and might not be utf8.

=> Fixed